scripts/missions/wotg/10_Daughter_of_a_Knight.lua

```diff
@@ -0,0 +1,266 @@
+                end,
+            },
+
+            onEventFinish =
+            {
+                [33] = function(player, csid, option, npc)
+                    player:setMissionStatus(mission.areaId, 4)
+                end,
+            },
+        },
+    },
+
+    -- 4. Examine it again to fight the Cernunnos NM.
+    --    Once Cernunnos is defeated, check the "Humus-rich Earth" again for a cutscene and Cernunnos Resin.
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId and missionStatus == 4
+        end,
+
+        [xi.zone.JUGNER_FOREST] =
+        {
+            ['Humus-rich_Earth'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getLocalVar("CERNUNNOS_DEFEATED") == 1 then
+                        -- TODO: This CS looks strange, the Treant is blocking the camera...
+                        return mission:progressEvent(34, 104, 300, 200, 100, 0, 1648, 0, 0) -- TODO: What is this?
+                    else
+                        local mob = GetMobByID(presentJugnerID.mob.CERNUNNOS)
+                        if not mob:isSpawned() then
+                            player:messageSpecial(presentJugnerID.text.DRAWN_UNWANTED_ATTENTION)
+                            SpawnMob(presentJugnerID.mob.CERNUNNOS):updateClaim(player)
+                        end
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [34] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.CERNUNNOS_RESIN)
+                    player:setLocalVar("CERNUNNOS_DEFEATED", 0)
+                    player:setMissionStatus(mission.areaId, 5)
+                end,
+            },
+        },
+    },
+
+    -- 5. Go back to Southern San d'Oria talk to Amaura for a cutscene.
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission == mission.missionId and missionStatus == 5
+        end,
+
+        [xi.zone.JUGNER_FOREST] =
+        {
+            ['Humus-rich_Earth'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:messageSpecial(presentJugnerID.text.DELIVER_TO_AMAURE)
+                end,
+            },
+        },
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Amaura'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(939, 0, 0, 0, 31, 0, 0, 1) -- TODO: What is this?
+                end,
+            },
+
+            onEventFinish =
+            {
+                [939] = function(player, csid, option, npc)
+                    player:delLeyItem(xi.ki.CERNUNNOS_RESIN)
+                    player:setMissionStatus(mission.areaId, 6)
+                    player:setCharVar("[WOTG10]Wait", getVanaMidnight())
```

Comment:
Since this seems to be internal to the mission itself, I'd suggest using `mission:setVar(player, 'Wait', getVanaMidnight())`, though another option instead of using midnight is storing VanadielUniqueDay() +1, and checking against that.

Reply:
👍 done